src/main/resources/db/reference/modernised_darts_schema.sql

```diff
@@ -795,12 +804,17 @@ IS 'primary key of judge';
 CREATE TABLE media
 (med_id                      INTEGER                       NOT NULL
 ,ctr_id                      INTEGER
+,met_id                      INTEGER                       NOT NULL
 ,media_object_id             CHARACTER VARYING(16)
 ,channel                     INTEGER                       NOT NULL -- 1,2,3,4 or rarely 5
 ,total_channels              INTEGER                       NOT NULL --99.9% are "4" in legacy, occasionally 1,2,5 
 ,reference_id                CHARACTER VARYING             --all nulls in legacy
 ,start_ts                    TIMESTAMP WITH TIME ZONE      NOT NULL
 ,end_ts                      TIMESTAMP WITH TIME ZONE      NOT NULL
+,media_file                  CHARACTER VARYING             NOT NULL
+,media_format                CHARACTER VARYING             NOT NULL
+,file_size                   INTEGER                       NOT NULL

```

Comment:
Should this column be `bigint`?

Reply:
assuming you are referring to "file_size" this is currently INTEGER, which has an upper limit of 2,147,483,647 and I believe is intended to record the file size in bytes.  This implies a maximum recorded file size of 2048Mb or 2Gb.
The largest recorded object in Legacy is 797,993,232 ( according to both dm_sysobject_s.r_content_size and dmr_content_s.full_content_size).  I'm not sure whether there is an upper limit on the size of the external objects ( inc media), but looks like we're currently at about 40% of what INTEGER can accommodate. What's the anticipated maximum size of media in bytes ?

Reply:
re-visiting in light of comment below, regarding media_type, i note that the datatype of the two fields in legacy database are defined in both dm_sysobject_s and dmr_content_s, with a content_size of type number(10) and a full_content_size of float(126).  Throughout the live data, both fields are populated, as no objects exceed the capacity of number(10).  However, to ensure we aren't introducing any future limitation, not present in legacy, I agree BIGINT is a better choice, but still far lower capacity than that offered by FLOAT(126).  That is 1e18 vs 1e38 ! But based on current usage should be more than adequate.